invenio_cli/commands/steps.py

```diff
@@ -10,35 +10,52 @@
 from ..helpers.process import run_interactive
 
 
-class FunctionStep(object):
+class Step(object):
+    """Interface for step objects."""
+
+    def __init__(self, message=None, skippable=False):
+        """Constructor."""
+        self.message = message
+        self.skippable = skippable
+
+    def execute(self):
+        """Execute the step."""
+        raise NotImplementedError
+
+
+class FunctionStep(Step):
     """A step which execution is a function call.
 
     Is composed of a function, arguments, and a message (feedback).
     """
 
-    def __init__(self, func, args=None, message=None):
+    def __init__(self, func, args=None, **kwargs):
         """Constructor."""
+        super().__init__(**kwargs)
         self.func = func
         self.args = args or {}
-        self.message = message
 
     def execute(self):
         """Execute the function with the given arguments."""
-        return self.func(**self.args)
+        response = self.func(**self.args)
+        if self.skippable:
+            reponse.warning = True
+            response.status_code = 0
```

Comment:
Question: what happens if `func` will return an error? Is it ok to set `response.status_code = 0` even it that case?

Reply:
It is "okay" because otherwise the logic of "error -> fail" vs "just show warning" becomes quite complex since not all subprocess/functions behave the same...